### PR TITLE
CSRF Policy should not block same-site requests

### LINKED ISSUE #15: Generic CSRF Policy blocks same-site requests
The generic CSRF policy should be updated to allow requests that have the `sec-fetch-site` [(MDN)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site) header with a value of `same-site` or `same-origin`.  Otherwise, it blocks the request that does not have, and does not need, a CSRF token.

### PR DESCRIPTION
**Pull request by joekhoobyar**
## what
* Updates the CSRF rule set to only match requests that are missing `sec-fetch-site` with value of `same-site` or `same-origin`.

## why
* The current CSRF rule set has false positives for compliant websites.

## references
* Fixes #15

### CODE REVIEW

**Review by exequielrafaela — APPROVED** [review]

File: modules/waf-regional/wafregional_ruleset8_csrf.tf
Comment (diego-ojeda-binbash) [inline modules/waf-regional/wafregional_ruleset8_csrf.tf:64]: @joekhoobyar  There is a lot of magic going here :) -- I mean, I know CloudFront uses its own conventions to accomplish several goals, I just think it would nice to have a single line that references the documentation that explains where the values in the "byte_match_tuples" block come from (target_string, data, etc).

**Review by diego-ojeda-binbash — APPROVED** [review]

File: modules/waf-regional/wafregional_ruleset8_csrf.tf
Comment (exequielrafaela) [reply modules/waf-regional/wafregional_ruleset8_csrf.tf:64]: @joekhoobyar Do you think that would be possible to attend @diego-ojeda-binbash request :point_up: before we merge this PR? Thanks in advance! :pray:

**Review by mpagnucco — APPROVED** [review]

**Review by exequielrafaela — CHANGES_REQUESTED** [review]
@joekhoobyar please when possible we'll wait for your feedback regarding this comment https://github.com/binbashar/terraform-aws-waf-owasp/pull/23#discussion_r484967955 in order to merge this PR. Thanks a lot! 

CC: @binbashar/leverage-project-terraform-admin

**Pull request merged**
